### DISCUSSION TITLE
Allow defining control minimum size in theme

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#5604] [GUI]
Many style guides recommend setting a minimum width for buttons, so buttons with short texts have a uniform look.

A negative example is:

![ksnip_20221017-100117](https://user-images.githubusercontent.com/372476/196073448-25ad8c20-8ced-4751-bd2e-e5580df2ea88.png)

We can currently set minimum width for individual controls, but not in theme. For styleboxes, we can only set content margin.

**Comment by Calinou** [comment]
> We can currently set minimum width for individual controls, but not in theme. For styleboxes, we can only set content margin.

Setting the content margin would probably be the way to go here, but I had to remove it while working on https://github.com/godotengine/godot/pull/45607 because it looked bad on icon-only buttons (which need to be square).

**Reply by timothyqiu** [reply]
How would content margin help? It only adds extra space in addition to existing text size. So the user still have to set different content margin for different text I think.

**Reply by Calinou** [reply]
I can notice what you're referring to in KDE's settings menu (notice how "Cannot be deleted" has smaller horizontal padding):

![image](https://user-images.githubusercontent.com/180032/196424980-3ef2deaa-0e7b-494a-8dc7-fd057588f351.png)

I can see what it's useful for, but I don't know how it could be done in Godot. The StyleBox's sizing *can* determine the overall control's clickable area and size, so it might be possible after all. If you can find a way to, feel free to open a pull request :slightly_smiling_face: 

___

Regarding improving the status quo without touching core:

In general, what you need for buttons is to have a greater margin on the sides than on the top/bottom (typically 2× to 2.5× the margin). Once you have that, setting a minimum width isn't so important anymore. It would be nice to have, but it's not critical for usability as all buttons become wider (except those that only use an icon, as I explained above). https://github.com/godotengine/godot-proposals/issues/4338 can also help here.

The issue is a bit like having a "default" style to highlight the main action in dialog buttons. It can be done already, but it requires a lot of work to adapt the editor to use the style only where needed.
